### PR TITLE
adding redis store

### PR DESCRIPTION
**Pull request by NewtTheWolf**

### CODE REVIEW

File: src/stores/RedisStore.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public async set(ip: string, ratelimit: Ratelimit) {
+        (await this.store).set(ip, JSON.stringify(ratelimit));
+    }
+
+    public async delete(ip: string) {
+        (await this.store).del(ip);
+    }
+
+    public async has(ip: string) {
+        const VALUE = (await this.store).exists(ip)
```
Comment (adiologydev) [inline src/stores/RedisStore.ts]: make `VALUE` small case and the `await` doesn't need to be inside `()` remove `()`.

File: src/stores/RedisStore.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            db: options.db,
+            password: options.password,
+            name: options.name,
+            maxRetryCount: options.maxRetryCount,
+            retryInterval: options.retryInterval,
+        })
+    }
+
+    public async get(ip: string) {
+        const data = await (await this.store).get(ip)
```
Comment (adiologydev) [inline src/stores/RedisStore.ts]: It only needs one `await` and the `await` doesn't need to be inside `()` remove `()`.

File: src/stores/RedisStore.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        if (!data) return
+        return JSON.parse(data)
+    }
+
+    public async set(ip: string, ratelimit: Ratelimit) {
+        (await this.store).set(ip, JSON.stringify(ratelimit));
```
Comment (adiologydev) [inline src/stores/RedisStore.ts]: The `await` doesn't need to be inside `()` remove `()`.

Comment (adiologydev) [reply src/stores/RedisStore.ts]: This must also return the new value as callback, you could pretty much just return this statement.

File: src/stores/RedisStore.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public async delete(ip: string) {
+        (await this.store).del(ip);
```
Comment (adiologydev) [inline src/stores/RedisStore.ts]: The `await` doesn't need to be inside `()` remove `()`.

Comment (adiologydev) [reply src/stores/RedisStore.ts]: Return this as well.

File: src/stores/RedisStore.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public async has(ip: string) {
+        const VALUE = (await this.store).exists(ip)
+        return !!await VALUE
```
Comment (adiologydev) [inline src/stores/RedisStore.ts]: No need for `await` or `!!` here.

**Review by adiologydev — CHANGES_REQUESTED** [review]
Do these changes and it should be good to go!

File: src/stores/RedisStore.ts (hunk 5 above)
Comment (NewtTheWolf) [reply src/stores/RedisStore.ts]: since redis returns a number (`0` or `1`) `!!` makes it a boolean

File: src/stores/RedisStore.ts (hunk 1 above)
Comment (NewtTheWolf) [reply src/stores/RedisStore.ts]: image
when i remove the brackets this error comes up

File: src/stores/RedisStore.ts (hunk 2 above)
Comment (NewtTheWolf) [reply src/stores/RedisStore.ts]: image
when i remove the brackets this error comes up

File: src/stores/RedisStore.ts (hunk 3 above)
Comment (NewtTheWolf) [reply src/stores/RedisStore.ts]: > The `await` doesn't need to be inside `()` remove `()`.

image
when i remove the brackets this error comes up

File: src/stores/RedisStore.ts (hunk 4 above)
Comment (NewtTheWolf) [reply src/stores/RedisStore.ts]: > The `await` doesn't need to be inside `()` remove `()`.

image
when i remove the brackets this error comes up

**Comment by adiologydev** [comment]
I am accepting these changes for now but there will be more changes I'll be making before releasing a new release candidate. Thanks for your contribution.

**Pull request merged**